scipy/sparse/_construct.py

```diff
@@ -66,10 +78,17 @@ def spdiags(data, diags, m=None, n=None, format=None):
         m = n = len(data[0])
     elif n is None:
         m, n = m
-    return dia_matrix((data, diags), shape=(m, n)).asformat(format)
+    D = dia_matrix((data, diags), shape=(m, n)).asformat(format)
+    if sparray is None:
+        sparray = False
```

Comment:
There's a question of whether we want to start warning about the future change here already, but since there is no timeline for deprecating matrices yet, I think this is fine.

Reply:
+1 I would rather get more infrastructure in place and some feedback from users before making a deprecation plan. But, hopefully, that will be coming soon (even if the actual process of removing sparse matrices takes time).